Sources/MapboxMaps/Foundation/MapboxMap.swift

```diff
@@ -78,21 +78,15 @@ public final class MapboxMap: MapboxMapProtocol {
     // MARK: - Style loading
 
     private func observeStyleLoad(_ completion: @escaping (Result<Style, Error>) -> Void) {
-        onNext(eventTypes: [.styleLoaded, .mapLoadingError]) { event in
-            switch event.type {
-            case MapEvents.styleLoaded:
-                if !self.style.isLoaded {
-                    Log.warning(forMessage: "style.isLoaded == false, was this an empty style?", category: "Style")
-                }
-                completion(.success(self.style))
-
-            case MapEvents.mapLoadingError:
-                let error = MapLoadingError(data: event.data)
-                completion(.failure(error))
-
-            default:
-                fatalError("Unexpected event type")
+        onTypedNext(.styleLoaded) { _ in
+            if !self.style.isLoaded {
+                Log.warning(forMessage: "style.isLoaded == false, was this an empty style?", category: "Style")
             }
+            completion(.success(self.style))
+        }
+
+        onTypedNext(.mapLoadingError) { event in
+            completion(.failure(event.payload.error))
```

Comment:
Not quite related to this PR, but I assume we have to save cancellables for this calls.

Reply:
I guess we could cancel the previous subscriptions when trying to load a new style before a previous one is loaded, e.g:
```
mapboxMap.loadStyle(.light) { _ in }
mapboxMap.loadStyle(.satellite) { _ in } <- this call would cancel the `styleLoaded` and `mapLoadingError` observations from the previous line
```